pygtrie.py

```diff
@@ -1364,13 +1364,35 @@ def add(self, key):
             self._trie[key:] = True
 
     def discard(self, key):
-        raise NotImplementedError(
-            'Removing keys from PrefixSet is not implemented.')
+        try:
+            node, _ = self._get_node(key)
+        except:
+            pass
+
+        if node and node.children:
```

Comment:
I don’t understand why you have this special case. ``node.children = {}`` is enough to remove all the children, you don’t have to iterate over them.